### PR TITLE
Add rule that checks if DynamoDB table enables server-side encryption…

### PR DESCRIPTION
**Pull request by dscpinheiro**
… using a CMK stored in KMS

### CODE REVIEW

**Comment by dscpinheiro** [comment]
Did you have a chance to look at this PR @erickascic?

File: lib/cfn-nag/custom_rules/DynamoDBEncryptionRule.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W74'
+  end
+
+  def audit_impl(cfn_model)
+    violating_ddb_tables = cfn_model.resources_by_type('AWS::DynamoDB::Table').select do |table|
+      table.sSESpecification.nil? || !truthy?(table.sSESpecification['SSEEnabled'].to_s)
```
Comment (erickascic) [inline lib/cfn-nag/custom_rules/DynamoDBEncryptionRule.rb:22]: the point of this rule is only to flag an AWS owned key v. AWS managed key? don't care about the KmsKeyId property?

Comment (dscpinheiro) [reply lib/cfn-nag/custom_rules/DynamoDBEncryptionRule.rb:22]: Yes, it's just to check that the key is not owned by Amazon DynamoDB.

Comment (erickascic) [reply lib/cfn-nag/custom_rules/DynamoDBEncryptionRule.rb:22]: philosophical question for you - do you think not specifying explicit customer CMK for resources should be a warning?  we have a few resources where it does warn... but there are about 40 other resources we don't check.  I was thinking to handle this consistently and make it a warning

Comment (dscpinheiro) [reply lib/cfn-nag/custom_rules/DynamoDBEncryptionRule.rb:22]: I talked to a co-worker and we agreed it probably shouldn't, as it depends a lot on whatever the customer needs are. For some, just using an AWS CMK is enough, but others might need to use a customer CMK for compliance reasons.

Comment (dscpinheiro) [reply lib/cfn-nag/custom_rules/DynamoDBEncryptionRule.rb:22]: I talked to a co-worker as well and we agreed it probably shouldn't be a warning, as it depends a lot on whatever the customer needs are. For some, just using an AWS CMK is enough, but others might need to use a customer CMK for further compliance reasons.

**Pull request merged**